runbotics/runbotics-scheduler/src/mail/mail.service.ts

```diff
@@ -0,0 +1,203 @@
+
+                    const process = await this.validateTitle(msg.envelope);
+                    const variables = await this.extractMailBody(msg.source);
+                    
+                    const input = { variables };
+
+                    await this.queueService.createInstantJob({
+                        process,
+                        input,
+                        user: null,
+                        trigger: ProcessTrigger.EMAIL,
+                        triggeredBy: sender.toLowerCase(),
+                    });
+                } catch (error) {
+                    this.logger.error(error.message, error);
+                    await this.sendReplyErrorMail(msg.envelope, error.message);
+                }
+            }
+
+            if (processedMails.length) {
+                this.logger.log('Mails processsed: ', processedMails);
+            } else {
+                this.logger.log('No mails processed ');
+            }
+
+            await this.client.messageFlagsAdd(processedMails, ['\\Seen']);
+        } finally {
+            await this.client.mailboxClose();
+            lock.release();
+            await this.client.logout();
+            this.logger.log('<< Mailbox disconnected');
+        }
+    }
+
+    private async validateTitle(envelope: MessageEnvelopeObject) {
+        const process = await this.processService.findByInfo(envelope.subject);
+
+        if (!process) {
+            this.logger.error(`Process "${envelope.subject}" does not exist`);
+            throw new NotFoundException(`Process "${envelope.subject}" does not exist`);
+        }
+
+        await this.queueService.validateProcessAccess({ process, triggered: true });
+
+        return process;
+    }
+
+    private sendReplyErrorMail(envelope: MessageEnvelopeObject, text: string) {
+        return this.server.sendMail({
+            from: envelope.to[0].address,
+            to: envelope.from[0].address,
+            replyTo: envelope.replyTo[0].address,
+            inReplyTo: envelope.messageId,
+            subject: envelope.subject,
+            text: `Error running process ${envelope.subject}\n\n${text}`,
+        });
+    }
+
+    private async extractMailBody(body: Buffer) {
+        try {
+            const parsed = await simpleParser(body);
+            const input = this.parseMailBody(parsed.text);
+            return {
+                ...input,
+                ...this.extractAttachments(parsed.attachments)
+            };
+        } catch (error) {
+            this.logger.error(`Wrong body format: ${error.message}`);
+            throw new Error(`Wrong body format: ${error.message}`);
+        }
+    }
+
+    private extractAttachments(attachments: Attachment[]) {
+        return attachments.reduce((acc, att) => {
+            const filename = att.filename.split('.').slice(0, -1).join('');
+            acc[filename] = `data:${att.contentType};base64,'${att.content.toString('base64')}`;
```

Comment:
Can we format it to our format `filename:<filename>;data:<content-type>;base64,<base64string>`?